### PR TITLE
add optimized `#fid_collections_by_type` query for Wings

### PR DESCRIPTION
**Pull request by no-reply**
this is follow-up to #4430, adding an optimized query to Wings. this should be vastly faster than the default query for the Wings adapter.

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: spec/wings/services/custom_queries/find_collections_by_type_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      let(:collections_with_target_type) do
+        FactoryBot.create_list(:collection_lw, 3, collection_type_gid: type_gid)
+      end
+
+      before do # force create all the models
+        non_collection_model
+        collection_with_default_type
+        collections_with_target_type
+      end
```
Comment (cjcolvar) [inline spec/wings/services/custom_queries/find_collections_by_type_spec.rb:31]: I'm curious of your thoughts on lazy loading these in a before block vs. using `let!`.

Comment (no-reply) [reply spec/wings/services/custom_queries/find_collections_by_type_spec.rb:31]: i think rubocop bars `let!` and this is my attempt at a slightly more intent-revealing approach for that.

Comment (cjcolvar) [reply spec/wings/services/custom_queries/find_collections_by_type_spec.rb:31]: Interesting.  I tend to think that `let!` is clear (except that the little `!` can be easily overlooked) and I usually find the simple references in a before block odd-looking but your comment makes it better by making the intent explicit.

**Comment by no-reply** [comment]
i fixed the failing test.

File: spec/wings/services/custom_queries/find_collections_by_type_spec.rb (hunk 1 above)
Comment (no-reply) [reply spec/wings/services/custom_queries/find_collections_by_type_spec.rb:31]: the issue with `let!`, i think, is that it leads folks to write test suites that eagerly create unneeded test objects. my understanding is that the style restriction seeks to make this deliberately more cumbersome/visible.

**Pull request merged**